pkgs/applications/backup/proxmox-backup-client/default.nix

```diff
@@ -37,9 +38,17 @@ rustPlatform.buildRustPackage rec {
   sourceRoot = "proxmox-backup";
 
   cargoPatches = [
-    ./0001-re-route-dependencies-not-available-on-crates.io-to-.patch
-    ./0002-docs-drop-all-but-client-man-pages.patch
-    ./0003-docs-Add-target-path-fixup-variable.patch
+    (fetchpatch {
+      name = "0001-re-route-dependencies-not-available-on-crates.io-to-.patch";
+      url = "https://aur.archlinux.org/cgit/aur.git/plain/0001-re-route-dependencies-not-available-on-crates.io-to-.patch?h=proxmox-backup-client&id=83a1f4dfcb04bd181b11954b1d9f5ddfcb72b3d0";
+      hash = "sha256-2YZtjbpYSbRk6rmpjKJeIO+V0YN5PrKsISONXMj4RG0=";
+    })
+    (fetchpatch {
+      name = "0002-docs-drop-all-but-client-man-pages.patch";
+      url = "https://aur.archlinux.org/cgit/aur.git/plain/0002-docs-drop-all-but-client-man-pages.patch?h=proxmox-backup-client&id=83a1f4dfcb04bd181b11954b1d9f5ddfcb72b3d0";
+      hash = "sha256-oJKQs4SwJvX5Zd0/l/vVr66aPO7Y4AC8byJHg9t1IhY=";
+    })
+    ./0001-docs-Add-target-path-fixup-variable.patch
```

Comment:
was there any effort made to upstream these patches?

Reply:
That's not really an option, at least for the first two.

The situation is basically the following:
- A lot of Rust crates it depends on are not published through crates.io (or through other means, for that matter), and are only available via git or Debian packages. The build system assumes that these crates are installed globally and thus available from a local crates registry. The first patch addresses this by simply changing them to use relative paths.
We could *theoretically* package all the crates separately and use them that way, but that would be more effort than it warrants. Maybe I will look into it if I have some spare time to dedicate to this. 
For now, either the patch or we don't package this at all.

- The second patch simply prevents the generation of some (unneeded) man pages, where their source file (a `.rst`) is generated as a separate build-step, depended on all project binaries to be compiled. Thus not strictly necessary, but it would be a waste of resources. 

The third patch, looking at it again, maybe - I'll see what I can do, but this shouldn't be a blocker for this PR.

Reply:
It off course isn't a blocker for the PR what I have people seen doing is adding in comments linking to pr's or stating that the patch can't be upstreamd which makes it easier for people maintaining the package in the future.